pm_muairss_write/pm_muairss_write.xml

```diff
@@ -1,29 +1,52 @@
     </requirements>
     <command detect_errors="exit_code"><![CDATA[
         pip show pymuon-suite || ( git clone https://github.com/muon-spectroscopy-computational-project/pymuon-suite /tmp/pymuon-suite && pip install /tmp/pymuon-suite && rm -rf /tmp/pymuon-suite) &&
-        cp $structure input.cell &&
-        cp $params params.yaml &&
+        ln -s $structure input.cell &&
+        ln -s $params params.yaml &&
+        #if str($iscastep.iscastep_sel)=="true":
+            ln -s $castep_param $castep_param.name &&
+        #end if
+        #if str($iscastep.iscastep_sel)=="true":
+            sed -i "s#^castep_param: .*#castep_param: .\/$castep_param.name#g" params.yaml &&
+        #end if
         pm-muairss -t w input.cell params.yaml &&
-        zip -r out_zip.zip $out_folder &&
-        find $out_folder > tree.txt
+        out_folder="`python ${__tool_directory__}/get_out_folder.py`" &&
+        #if str($iscastep.iscastep_sel)=="true":
+            zip -r out_zip.zip \$out_folder params.yaml input.cell $castep_param.name &&
+        #else if str($iscastep.iscastep_sel)=="false":
+            zip -r out_zip.zip \$out_folder params.yaml input.cell &&
+        #end if
+        find \$out_folder > tree.txt
     ]]></command>
     <inputs>
         <param type="data" name="structure" format="cell" />
-        <param type="data" name="params" format="yaml" />
-        <param type="text" name="out_folder"/>
+        <param type="data" name="params" format="txt" />
+    <conditional name="iscastep">
+        <param type="boolean" value="false" name="iscastep_sel" label="Is the calculator castep?">
+            <option value="false">no</option>
+            <option value="true">yes</option>
+        </param>   
+        <when value="true">
+            <param type="data" name="castep_param" format="txt" />
+        </when>
+    </conditional>
     </inputs>
     <outputs>
-        <data label="$out_folder" name="out_zip" format="zip" from_work_dir="out_zip.zip"/>
+        <data label="muonated $structure.name using $params.name" name="out_zip" format="zip" from_work_dir="out_zip.zip"/>
         <data label="file tree for testing" name="file_tree" format="txt" from_work_dir="tree.txt" hidden="true"/>
     </outputs>
     <tests>
         <test>
             <param name="structure" value="Si.cell" ftype="txt"/>
-            <param name="params" value="Si-muairss-castep.yaml" ftype="txt"/>
-            <param name="out_folder" value="si-muon-airss-out"/>
-            <output name="out_zip" file="si-muon-airss-out.zip" ftype="zip" compare="sim_size" delta_frac="0.1"/>
+            <param name="params" value="Si-muairss-castep-with-out-folder.yaml" ftype="txt"/>
+            <output name="file_tree" file="tree.txt" ftype="txt" compare="sim_size" delta="350"/>
+        </test>
+        <test>
+            <param name="structure" value="Si.cell" ftype="txt"/>
+            <param name="params" value="Si-muairss-castep-no-out-folder.yaml" ftype="txt"/>
```

Comment:
One thing that we miss in these tests is whether the castep_param variable is correctly updated/whether the generated param files have the expected contents. Can you think of any way to test this?

Reply:
Discussed: this is a difficult test to make, and would require writing more content to test files as well even if we figured out how to approach it. We will ask the Galaxy developers if there is a way to run a script as part of testing. If there isn't, we will rely on manual testing for this aspect.

Reply:
Resolved: use hidden parameters to write extra stuff to file